source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -352,7 +354,8 @@ Filter::StreamOpenState Filter::openStream() {
     }
     stats_.streams_started_.inc();
 
-    stream_ = config_->threadLocalStreamManager().store(this, std::move(stream_object));
+    stream_ = config_->threadLocalStreamManager().store(decoder_callbacks_->streamId(),
```

Comment:
Come to think about this more, stream_id might not the best choice as well. While the probability of key collision is very low given that it is a 64 bit random number, I think conceptually it would be best not to have it as an issue.

Please leave to TODO here to address this. It may best to use the address of the stream object as the key, since you only need to for deletion and deletion will do both, remove the key and delete the object.

Reply:
Make sense. We probably should encapsulate into threadLocalStreamManager to get the address of final moved object.

Done.